src/ParallelAlgorithms/LinearSolver/Multigrid/Tags.hpp

```diff
@@ -43,6 +43,33 @@ struct OutputVolumeData {
   static bool suggested_value() { return false; }
 };
 
+template <typename OptionsGroup>
+struct EnablePreSmoothing {
+  static std::string name() { return "PreSmoothing"; }
+  using type = bool;
+  static constexpr Options::String help =
+      "Set to 'False' to disable pre-smoothing altogether (\"cascading "
+      "multigrid\"). Note that pre-smoothing can be necessary to remove "
+      "high-frequency modes in the data that get restricted to coarser grids, "
+      "since such high-frequency modes can introduce aliasing. However, when "
+      "running only a single V-cycle as preconditioner, the initial field is "
+      "typically zero, so pre-smoothing may not worthwile.";
+  using group = OptionsGroup;
+};
+
+template <typename OptionsGroup>
+struct EnablePostSmoothingAtBottom {
+  static std::string name() { return "PostSmoothingAtBottom"; }
+  using type = bool;
+  static constexpr Options::String help =
+      "Set to 'False' to skip post-smoothing on the coarsest grid. This means "
+      "only pre-smoothing runs on the coarsest grid, so the coarsest grid "
+      "experiences less smoothing altogether. This is typically only "
+      "desireable if the coarsest grid covers the domain with a single "
+      "element, or very few, so smoothing is exceptionally effective.";
```

Comment:
"This is typically only desirable..." is "this" setting the option or setting the option to false (as discussed in the previous sentences)?  It seems to me to mean "Setting this option to false is typically only desirable", but then it seems weird to disable smoothing if "smoothing is exceptionally effective".

Reply:
expanded the text a bit

Reply:
That clears it up.  Thanks.